### PR TITLE
hyundai longitudinal safety

### PR DESCRIPTION
**Pull request by gregjhogan**
copied logic from honda bosch longitudinal safety
* same accel limits
* currrently checking aReqRaw for limits do we need to also check aReqValue?
* brought over `current_controls_allowed` logic (pedal + moving) not sure why this wasn't already there
* used safety param value of 1 to enable

### CODE REVIEW

**Comment by gregjhogan** [comment]
this is probably good to merge now

File: board/safety/safety_hyundai.h [hunk 1]
```diff
@@ -252,6 +288,7 @@ static void hyundai_legacy_init(int16_t param) {
   relay_malfunction_reset();
 
   hyundai_legacy = true;
+  hyundai_longitudinal = false;
```
Comment (adeebshihadeh) [inline board/safety/safety_hyundai.h:291]: why not allow long for the legacy safety mode?

Comment (gregjhogan) [reply board/safety/safety_hyundai.h:291]: addresses could be completely different and I have no way to test, what I have now I know is sufficient at least for my car to not have any errors (which I didn't get on the first try)

Comment (adeebshihadeh) [reply board/safety/safety_hyundai.h:291]: hyundai legacy is the same as the regular hyndai safety mode, except with a subset of the counter/checksum checks

Comment (gregjhogan) [reply board/safety/safety_hyundai.h]: ok, I added it

**Comment by gregjhogan** [comment]
I take it back, it isn't ready yet ...
* SCC12 RX check can only happen for stock longitudinal (since radar sends it).  We could add two more safety modes instead of using the safety param (since RX check list isn't currently dynamic), make RX check list dynamic, we might be able to switch to always using TCS13->ACC_REQ, or switch to always using button press detection for enable/disable.
* How do we feel about using button press detection for setting `controls_allowed` when it doesn't have a checksum?

**Comment by gregjhogan** [comment]
This can be finished whenever a decision is made on https://github.com/commaai/panda/pull/625

**Pull request closed without merging**